third-party/crowdsec/src/main/java/org/apache/james/model/CrowdsecHttpClient.java

```
@@ -0,0 +1,99 @@
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.model;
+
+import static org.apache.james.model.CrowdsecClientConfiguration.DEFAULT_TIMEOUT;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import javax.inject.Inject;
+
+import org.apache.james.util.AuditTrail;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+
+import reactor.core.publisher.Mono;
+import reactor.netty.http.client.HttpClient;
+
+public class CrowdsecHttpClient {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CrowdsecHttpClient.class);
+    private static final String GET_DECISION = "/decisions";
+
+    public static final int OK = 200;
+    public static final int UNAUTHENTICATED = 403;
+    public static final int NOT_FOUND = 404;
+
+    private final HttpClient httpClient;
+    private final ObjectMapper mapper;
+
+    @Inject
+    public CrowdsecHttpClient(CrowdsecClientConfiguration configuration) {
+        this.httpClient = buildReactorNettyHttpClient(configuration);
+        this.mapper = new ObjectMapper().registerModule(new Jdk8Module());
+    }
+
+    public Mono<List<CrowdsecDecision>> getCrowdsecDecisions() {
+        return httpClient.get()
+            .uri(GET_DECISION)
+            .responseSingle((response, body) -> {
+                switch (response.status().code()) {
+                    case OK:
+                        return body.asString().map(this::parseCrowdsecDecisions);
+                    case UNAUTHENTICATED:
+                        AuditTrail.entry()
+                            .log("Invalid api-key bouncer");
+                        return Mono.empty();
```

Please propagate the error in the client. Let the caller manage the error.

Error management is a hard topic, here we are taking technical decision that have a huge security/availability impact. Here is not the right place to take a decision.